src/customer/strategies/masterpass-customer-strategy.ts

```diff
@@ -0,0 +1,127 @@
+import CustomerCredentials from '../customer-credentials';
+import { CustomerInitializeOptions, CustomerRequestOptions } from '../customer-request-options';
+
+import CustomerStrategy from './customer-strategy';
+
+export default class MasterpassCustomerStrategy extends CustomerStrategy {
+    private _signInButton?: HTMLElement;
+    private _paymentMethod?: PaymentMethod;
+
+    constructor(
+        store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _remoteCheckoutActionCreator: RemoteCheckoutActionCreator,
+        private _masterpassScriptLoader: MasterpassScriptLoader
+    ) {
+        super(store);
+    }
+
+    initialize(options: CustomerInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { masterpass: masterpassOptions, methodId } = options;
+
+        if (!masterpassOptions || !methodId) {
+            throw new InvalidArgumentError('Unable to proceed because "options.masterpass" argument is not provided.');
+        }
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId))
+            .then(state => {
+
+                this._paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+                if (!this._paymentMethod || !this._paymentMethod.initializationData.checkoutId) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                const cart = state.cart.getCart();
+                if (!cart) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCart);
+                }
+
+                const { container } = masterpassOptions;
+
+                const payload = {
+                    checkoutId: this._paymentMethod.initializationData.checkoutId,
+                    allowedCardTypes: this._paymentMethod.initializationData.allowedCardTypes,
+                    amount: cart.cartAmount.toString(),
+                    currency: cart.currency.code,
+                    cartId: cart.id,
+                };
+
+                return this._masterpassScriptLoader.load(this._paymentMethod.config.testMode)
+                    .then(Masterpass => {
+                        this._signInButton = this._createSignInButton(container);
+
+                        this._signInButton.addEventListener('click', () => {
+                            Masterpass.checkout(payload);
+                        });
+                    });
+            })
+            .then(() => super.initialize(options));
+    }
+
+    deinitialize(options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!this._isInitialized) {
+            return super.deinitialize(options);
+        }
+
+        this._paymentMethod = undefined;
+        if (this._signInButton && this._signInButton.parentNode) {
+            this._signInButton.onclick = () => {};
```

Comment:
I think if you're removing the element, you don't need to remove the event listener yourself. The browser should be able to garbage collect and clean up the listeners for you once the element is removed from the DOM tree (and no lingering references). But if you want to do it for older browsers (or some other edge cases), you should use `removeEventListener` and explicitly remove the callback. This means not using an anonymous callback. But as I said, I'm happy for you to keep it simple and just remove this line.